torch_geometric/loader/link_neighbour_loader.py

```diff
@@ -0,0 +1,95 @@
+from typing import Any, Callable, List, Tuple
+
+import torch
+from torch import Tensor
+
+from torch_geometric.data import Data
+from torch_geometric.loader.base import BaseDataLoader
+from torch_geometric.loader.neighbor_loader import NeighborSampler
+from torch_geometric.loader.utils import filter_data
+from torch_geometric.typing import InputEdges, NumNeighbors, OptTensor
+from torch_geometric.utils.mask import mask_to_index
+
+
+class LinkNeighborSampler(NeighborSampler):
+    def __init__(
+        self,
+        data: Data,
+        num_neighbors: NumNeighbors,
+        replace: bool = False,
+        directed: bool = True,
+    ):
+        super().__init__(data, num_neighbors, replace, directed)
+        self.data = data
```

Comment:
It is pretty important to not save `data` here (as otherwise the data needs to be shared or replicated across `num_workers` during data loading). We could construct `node_index` directly from `self.colptr` and `self.row`.

Reply:
Good point!